File: AssTasks.Server/Models/TaskParent.cs
namespace AssTasks.Server.Models
{
    public class TaskParent
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public int Priority { get; set; }
        public int Frequency { get; set; }
        public int FrequencyType { get; set; }
        public List<int>? Days { get; set; }
        public bool Active { get; set; }
        public DateTime CreatedAt { get; set; }
        public ICollection<AssTask> AssTasks { get; } = new List<AssTask>();
    }
}

File: AssTasks.Client/src/app/models/task-parent.ts
import { TaskTypes } from '../constants/task.constants';

interface BaseTaskParent {
    id: number;
    title: string;
    description?: string;
    active: boolean;
    priority: number;
    frequency: number;
    createdAt: Date;
}

interface IntervalTaskParent extends BaseTaskParent {
    frequencyType: TaskTypes.INTERVAL_TASK;
}

interface DaysTaskParent extends BaseTaskParent {
    frequencyType: TaskTypes.DAYS_TASK;
    days: number[];
}

export type TaskParent = IntervalTaskParent | DaysTaskParent;